lib/colorado_lottery.rb

```diff
@@ -0,0 +1,33 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners, :current_contestants
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(name, game)
+    if name.age < 18
+      false
+    elsif !name.game_interests.include?(game.name)
+      false
+    else
+      true
+    end
+  end
+
+  def can_register?(name, game)
+    if interested_and_18?(name, game)
```

Comment:
We needed both the condition of `interested_and_18?` and one of the conditions of `!name.out_of_state?` OR `game.national_drawing?`. The correct implementation would look like: 
```
interested_and_18?(contestant, game) &&
    (game.national_drawing? || !contestant.out_of_state?)
```